src/plan/generational/immix/gc_work.rs

```diff
@@ -0,0 +1,186 @@
+use super::global::GenImmix;
+use crate::plan::CopyContext;
+use crate::plan::PlanConstraints;
+use crate::policy::space::Space;
+use crate::scheduler::gc_work::*;
+use crate::scheduler::GCWorkerLocal;
+use crate::util::alloc::Allocator;
+use crate::util::alloc::ImmixAllocator;
+use crate::util::opaque_pointer::*;
+use crate::util::{Address, ObjectReference};
+use crate::vm::*;
+use crate::AllocationSemantics;
+use crate::MMTK;
+use std::ops::{Deref, DerefMut};
+
+/// Copy context for generational immix. We include two copy allocators for the same immix space.
+/// We should use the defrag copy allocator for full heap GC with defrag, or the normal copy allocator
+/// for other GCs.
+pub struct GenImmixCopyContext<VM: VMBinding> {
+    plan: &'static GenImmix<VM>,
+    copy: ImmixAllocator<VM>,
+    defrag_copy: ImmixAllocator<VM>,
```

Comment:
I am not completely sure. I was following the pattern here: https://github.com/JikesRVM/JikesRVM/blob/0b6002e7d746a829d56c90acfc4bb5c560faf634/MMTk/src/org/mmtk/plan/generational/immix/GenImmixCollector.java#L63

The difference between the two allocators are the `copy` parameter. The defrag copy allocator has `copy = true` and is only used for defrag collection. `copy` is then passed to the immix space to initialise block initial state and to update `available_clean_pages_for_defrag`. However, the immix space actually knows whether it is in a defrag collection, and it seems no point to pass `copy` from the allocator. So I am actually not sure. 

It may work if we would like to use just one copying allocator here, but we need to change immix allocator/space. That is not the purpose of this PR.

Reply:
Maybe @steveblackburn or @caizixian know more about the design here.

Reply:
Yeah, that is what I meant. However, I don't think we want to do that refactoring in this PR. 1. The PR is about GenImmix, refactoring existing Immix allocator/space is not the purpose of this PR, 2. There might be reasons why it was implemented in this way in Java MMTk that we do not know (we haven't ported all Immix plans yet - missing sticky immix and RC immix).